pkg/core/db/sql_executor.go

```diff
@@ -75,7 +82,18 @@ func (e *SQLExecutor) QueryRowWith(sqlizeri sq.Sqlizer) (*sqlx.Row, error) {
 	}
 	sql, args, err := sqlizeri.ToSql()
 	if err != nil {
+		if isWriteConflict(err) {
+			panic(ErrWriteConflict)
+		}
 		return nil, errors.WithDetails(err, errors.Details{"sql": errors.SafeDetail.Value(sql)})
 	}
 	return db.QueryRowxContext(e.context, sql, args...), nil
 }
+
+func isWriteConflict(err error) bool {
+	var pqErr *pq.Error
+	if errors.As(err, &pqErr) {
+		return pqErr.Code == "40001" || pqErr.Code == "40P01"
```

Comment:
Just curious. How did you find out these are the error codes we want to detect?

Reply:
These are the error codes corresponded to the two errors mentioned in [documentation page on isolation level](https://www.postgresql.org/docs/9.5/transaction-iso.html).

Reply:
40P01 is not mentioned in that page. Where can I find it?

Reply:
You can find the error codes in https://www.postgresql.org/docs/9.1/errcodes-appendix.html. 40P01 is deadlock error, which is may indicate conflicts between running transactions.

Reply:
Can you include these in the code as comments?